hw6/Names.java

```diff
@@ -0,0 +1,109 @@
+import java.io.*;
+import java.util.*;
+import java.awt.*;
+
+public class Names {
+   public static final int numOfDecades   = 14;
+   public static final int startingYear   = 1880;
+   public static final int widthPerDecade = 70;
+   
+   public static void main(String[] args) throws FileNotFoundException {
+      Scanner input = new Scanner(new File("names.txt"));
+      Scanner console = new Scanner(System.in);
+      giveIntro();
+      //form searchCombo with name+space+sex (uppercase), read from user output.
+      System.out.print("name? ");
+      String name1 = console.nextLine();
+      String name = (name1.substring(0, 1).toUpperCase() + name1.substring(1));
+      System.out.print("sex (M or F)? ");
+      String sex = console.nextLine().toUpperCase();
+      String searchCombo = name + " " + sex;
+      //find searchCombo in input file. If there is a matching output, construct a drawing panel.
+      //If there is no matching output, print not found message.
+      String line = find(input, searchCombo);
+      if (line.length() > 0) {
+         drawGraph (line,searchCombo);
+      } else {
+         System.out.println("name/sex combination not found");
+      }
+   }   
+   // introduces the purpose of the program to the user
+   public static void giveIntro() {
+      System.out.println("This program allows you to search through the");
+      System.out.println("data from the Social Security Administration");
+      System.out.println("to see how popular a particular name has been");
+      System.out.println("since " + startingYear + " .");
+      System.out.println();
+   }   
+   // searches for formatted data that starts with searchCombo
+   // return line if found; return empty if not found
+   public static String find(Scanner input, String searchCombo) {
+      while (input.hasNextLine()) {
+         String line = input.nextLine();
+         //format token, and compare token based with search combo
+         String dataCombo = tokenFormat(line);
+         if (dataCombo.startsWith(searchCombo)) {
+            return line;
+         }
+      }
+      return "";
+   }  
+   //format data as datacombo to match searchcombo
+   public static String tokenFormat(String line) {
+      Scanner data = new Scanner(line);
+      String nameFromData = data.next();
+      String sexFromData = data.next();
+      String dataCombo = nameFromData + " " + sexFromData;
+      return dataCombo;
+   }   
+   //construct a DrawingPanel, passing line as parameter
+   public static void drawGraph (String line, String searchCombo) {
+      DrawingPanel panel = new DrawingPanel(widthPerDecade*14, 550);
```

Comment:
Please add space before and after `*` to make the code easier to read, and this applies to rest of the method.